From 923b5553807323f43957d30689d351ca70011181 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Mon, 6 Feb 2023 22:40:48 +0800
Subject: [PATCH 025/500] [Huawei] Revert "io_uring: fix soft lockup when call
 __io_remove_buffers"

Offering: HULK
hulk inclusion
category: feature
bugzilla: 188335

-------------------------------

This reverts commit 4222bec09a7e18d6316495401daafec746925a73.

We need to apply patch 788d0824269bef (io_uring: import 5.15-stable
io_uring) to move io_uring to separate directory and solve
the problem of CVE-2023-0240.
This patch can be reverted since patch 788d0824269bef contains it.

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
---
 fs/io_uring.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/io_uring.c b/fs/io_uring.c
index 6bcc80fd624e..d476fe1cc681 100644
--- a/fs/io_uring.c
+++ b/fs/io_uring.c
@@ -3946,7 +3946,6 @@ static int __io_remove_buffers(struct io_ring_ctx *ctx, struct io_buffer *buf,
 		kfree(nxt);
 		if (++i == nbufs)
 			return i;
-		cond_resched();
 	}
 	i++;
 	kfree(buf);
-- 
2.17.1

